python/tk_desktop/systray.py

```diff
@@ -51,25 +51,35 @@ class ApplicationEventFilter(QtCore.QObject):
         def __init__(self, window, parent=None):
             QtCore.QObject.__init__(self, parent)
             self._window = window
+            self._deactivate = False
 
         def eventFilter(self, obj, event):
-            if event.type() == QtCore.QEvent.ApplicationDeactivate:
-                # When the app loses focus and is in pinned mode, we hide the dialog automatically
-                # and move the app to the background so there's no more icon in the tray.
-                if self._window.state == SystrayWindow.STATE_PINNED:
-                    self._window.hide()
-                    if osutils is not None:
-                        osutils.make_app_background()
-            elif event.type() == QtCore.QEvent.ApplicationActivate:
-                # When the app gains focus and is in pinned mode, we bring the app to the background.
-                # Note that we are not showing the main dialog because there are multiple top levels
-                # windows that would have cause the application to activate.
-                if self._window.state == SystrayWindow.STATE_PINNED:
-                    if osutils is not None:
-                        osutils.make_app_foreground()
+            if not self._deactivate:
+                if event.type() == QtCore.QEvent.ApplicationDeactivate:
+                    # When the app loses focus and is in pinned mode, we hide the dialog automatically
+                    # and move the app to the background so there's no more icon in the tray.
+                    if self._window.state == SystrayWindow.STATE_PINNED:
+                        self._window.hide()
+                        if osutils is not None:
+                            osutils.make_app_background()
+                elif event.type() == QtCore.QEvent.ApplicationActivate:
+                    # When the app gains focus and is in pinned mode, we bring the app to the background.
+                    # Note that we are not showing the main dialog because there are multiple top levels
+                    # windows that would have cause the application to activate.
+                    if self._window.state == SystrayWindow.STATE_PINNED:
+                        if osutils is not None:
+                            osutils.make_app_foreground()
```

Comment:
not sure, but would it be better to call the base class first? Not sure what it does in the case, but if it does something related to activations, it could shadow the logic in this method.